### PR TITLE
refactor(Pagination): rewrite with hook

### PR DESCRIPTION
**Pull request by hengkx**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  refactor with hook         |
| 🇨🇳 Chinese |  用 react hook 重构         |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/pagination/Pagination.tsx [hunk 1]
```diff
@@ -123,48 +131,33 @@ export default class Pagination extends React.Component<PaginationProps, {}> {
...
-        }}
-      </ConfigConsumer>
+    return (
+      <RcPagination
+        {...restProps}
+        prefixCls={prefixCls}
+        selectPrefixCls={selectPrefixCls}
+        {...getIconsProps()}
+        className={extendedClassName}
+        selectComponentClass={isSmall ? MiniSelect : Select}
```
Comment (hengkx) [inline components/pagination/Pagination.tsx:150]: 注入属性不行吗？为什么要加个`MiniSelect`

Comment (yoyo837) [reply components/pagination/Pagination.tsx:150]: 应该是太久没动了，我觉得可以改

Comment (yoyo837) [reply components/pagination/Pagination.tsx:150]: 它这里传入的是class 不是实例

Comment (hengkx) [reply components/pagination/Pagination.tsx:150]: 怎么感觉 有点怪怪的

File: components/pagination/Pagination.tsx [hunk 2]
```diff
@@ -53,28 +53,36 @@ export interface PaginationConfig extends PaginationProps {
...
+        forceUpdate();
       }
     });
-  }
-
-  componentWillUnmount() {
-    ResponsiveObserve.unsubscribe(this.token);
-  }
+    return () => {
+      ResponsiveObserve.unsubscribe(token);
```
Comment (afc163) [inline components/pagination/Pagination.tsx:79]: 看看能不能用 https://github.com/ant-design/ant-design/pull/22226

Comment (hengkx) [reply components/pagination/Pagination.tsx]: 用了，为啥 名字 叫`useBreakpoint`

**Comment by hengkx** [comment]
image
这个ci是什么鬼

File: components/pagination/Pagination.tsx (hunk 2 above)
Comment (afc163) [reply components/pagination/Pagination.tsx]: https://github.com/ant-design/ant-design/issues/13785#issuecomment-516725451

选了一个。

Comment (hengkx) [reply components/pagination/Pagination.tsx]: 我看不懂

**Comment by afc163** [comment]
要加个 `// deps-lint-skip: grid`。

**Comment by hengkx** [comment]
意思是我使用了grid里面的 组件？

**Comment by afc163** [comment]
对，但是没用到样式，需要声明忽略掉。

**Pull request merged**